lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,771 @@
+
+# Have any of them still not been processed?
+# (no CC_dark or CC-miss sections in _stats.json)
+        #[print(run) for run in (list_unprocessed_runs(Path,rse))]
+        np_runs = list_unprocessed_runs(Path, rse)
+        print(len(np_runs), " unprocessed runs found for this RSE")
+
+# Was the latest run ever attempted to be processed?
+
+        latest_run = list_runs(Path, rse, 1)[0]
+        print("Was the latest run", latest_run, "attempted to be processed already? ",\
+          was_cc_attempted(latest_run))
+        if was_cc_attempted(latest_run) is False or force_proceed is True:
+            print("Will try to process the run")
+
+###
+# Address the Dark files first
+###
+
+# Is there another run, at least "minagedark" old, for this RSE?
+            oldenough_run = None
+            max_files_at_site = 0
+            #print(list_runs_by_age(Path,rse, latest_run))
+            d = list_runs_by_age(Path, rse, latest_run)
+            if len([k for k in d if d[k] > minagedark]) > 0:\
+              # i.e. there is another dark run with appropriate age
+                oldenough_run = [k for k in d if d[k] > minagedark][0]
+                print("Found another run,", minagedark,\
+                  "days older than the latest!\nWill compare the dark files in the two.")
+                print("The first", minagedark, "days older run is: ", oldenough_run)
+
+# Create a cc_dark section in the stats file
+
+                t0 = time.time()
+                stats_key = "cc_dark"
+                cc_stats = stats = None
+                stats = Stats(latest_run)
+                cc_stats = {
+                    "start_time": t0,
+                    "end_time": None,
+                    "initial_dark_files": 0,
+                    "confirmed_dark_files": 0,
+                    "x-check_run": oldenough_run,
+                    "status": "started"
+                }
+                stats[stats_key] = cc_stats
+
+# Compare the two lists, and take only the dark files that are in both
+                latest_dark = re.sub('_stats.json$', '_D.list', latest_run)
+                oldenough_dark = re.sub('_stats.json$', '_D.list', oldenough_run)
+                print("\nlatest_dark =", latest_dark)
+                print("oldenough_dark =", oldenough_dark)
+                confirmed_dark = "%s_DeletionList.csv" % latest_run
+                confirmed_dark = re.sub('_stats.json$', '_DeletionList.csv', latest_run)
+                cmp2dark(new_list=latest_dark, old_list=oldenough_dark,\
+                  comm_list=confirmed_dark, stats_file=latest_run)
+
+###
+#   SAFEGUARD
+#   If a large fraction (larger than 'maxdarkfraction') of the files at a site
+#   are reported as 'dark', do NOT proceed with the deletion.
+#   Instead, put a warning in the _stats.json file, so that an operator can have a look.
+###
+
+# Get the number of files recorded by the scanner
+                print("latest_run", latest_run)
```

Comment:
Remove the print statements or use the logging modules common to Rucio

Reply:
Will do!

Reply:
Done.
The only print statements that remain are the ones from Igor's 'Stats' class. These can stay in for now.